bounter/count_min_sketch.py

```diff
@@ -103,144 +65,74 @@ def __init__(self, size_mb=64, width=None, depth=None, algorithm='conservative')
             self.width = width
             self.depth = depth
 
-        self.logshift = 0
-        self.inc_method = self.conservative_increment
-        if algorithm and str(algorithm).lower() == 'basic':
-            self.inc_method = self.basic_increment
-        if algorithm and str(algorithm).lower() == 'logcounter8':
-            self.logshift = 2
-            self.inc_method = self.log_increment
-            self.decode = self.log_decode
-        if algorithm and str(algorithm).lower() == 'logcounter1024':
-            self.logshift = 9
-            self.inc_method = self.log_increment
-            self.decode = self.log_decode
-        if algorithm and str(algorithm).lower() == 'logcons1024':
-            self.logshift = 9
-            self.inc_method = self.log_increment_conservative
-            self.decode = self.log_decode
-
-        self.hash_length = self.width.bit_length()
-        self.hash_mask = (1 << (self.hash_length - 1)) - 1
-        self.logbase = 1 << (self.logshift + 1)
-        self.table = numpy.zeros((self.depth, self.width), self.array_type)
-
-        self.sum = 0
-        self.distinct_counter = HyperLogLog(15)
+        if algorithm and str(algorithm).lower() == 'log8':
+            self.cms = CMSC.CMS_Log8(width=self.width, depth=self.depth)
```

Comment:
Should it works **only** with C "backend"? 
For example, in gensim we have both (python and cython variants), if cython don't compiled, by default, we used python version.

CC: @piskvorky

Reply:
I would say C only is OK. Maintaining both versions is Bounter is not required.